File: sample/OpenCvSharp.Samples/Samples/Blob.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using OpenCvSharp.Blob; // for blob

namespace OpenCvSharp.Test
{
    /// <summary>
    /// c# implementation of cvblob/test/test.cpp.
    /// cvblob : http://code.google.com/p/cvblob/
    /// </summary>
    class Blob 
    {
        public Blob()
        {
            using (IplImage imgSrc = new IplImage(Const.ImageShapes, LoadMode.Color))
            using (IplImage imgBinary = new IplImage(imgSrc.Size, BitDepth.U8, 1))
            using (IplImage imgLabel = new IplImage(imgSrc.Size, BitDepth.F32, 1))            
            using (IplImage imgRender = new IplImage(imgSrc.Size, BitDepth.U8, 3))
            using (IplImage imgContour = new IplImage(imgSrc.Size, BitDepth.U8, 3))
            using (IplImage imgPolygon = new IplImage(imgSrc.Size, BitDepth.U8, 3))
            {
                Cv.CvtColor(imgSrc, imgBinary, ColorConversion.BgrToGray);
                Cv.Threshold(imgBinary, imgBinary, 100, 255, ThresholdType.Binary);

                using (CvBlobs blobs = new CvBlobs())
                {
                    uint result = blobs.Label(imgBinary, imgLabel);

                    foreach (KeyValuePair<uint, CvBlob> item in blobs)
                    {
                        CvBlob b = item.Value;
                        Console.WriteLine("{0} | Centroid:{1} Area:{2}", item.Key, b.Centroid, b.Area);

                        CvContourChainCode cc = b.Contour;
                        cc.RenderContourChainCode(imgContour);

                        CvContourPolygon polygon = cc.ConvertChainCodesToPolygon();
                        foreach (CvPoint p in polygon)
                        {
                            imgPolygon.Circle(p, 1, CvColor.Red, -1);
                        }
                    }

                    blobs.RenderBlobs(imgLabel, imgSrc, imgRender);

                    using (new CvWindow("render", imgRender))
                    using (new CvWindow("contour", imgContour))
                    using (new CvWindow("polygon vertices", imgPolygon))
                    {
                        Cv.WaitKey(0);
                    }
                }

            }
        }
    }
}

File: sample/OpenCvSharpSamplesVB/Samples/Blob.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports OpenCvSharp
Imports OpenCvSharp.Blob ' for blob

' Namespace OpenCvSharpSamplesVB
    ''' <summary>
    ''' c# implementation of cvblob/test/test.cpp.
    ''' cvblob : http://code.google.com/p/cvblob/
    ''' </summary>
    Friend Module Blob
        Public Sub Start()
            Using imgSrc As New IplImage([Const].ImageShapes, LoadMode.Color)
                Using imgBinary As New IplImage(imgSrc.Size, BitDepth.U8, 1)
                    Using imgLabel As New IplImage(imgSrc.Size, BitDepth.F32, 1)
                        Using imgRender As New IplImage(imgSrc.Size, BitDepth.U8, 3)
                            Using imgContour As New IplImage(imgSrc.Size, BitDepth.U8, 3)
                                Using imgPolygon As New IplImage(imgSrc.Size, BitDepth.U8, 3)
                                    Cv.CvtColor(imgSrc, imgBinary, ColorConversion.BgrToGray)
                                    Cv.Threshold(imgBinary, imgBinary, 100, 255, ThresholdType.Binary)

                                    Dim blobs As New CvBlobs()
                                    blobs.Label(imgBinary)

                                    For Each item As KeyValuePair(Of Integer, CvBlob) In blobs
                                        Dim b As CvBlob = item.Value

                                        Form1.TextBox1.AppendText(Environment.NewLine & String.Format("{0} | Centroid:{1} Area:{2}", item.Key, b.Centroid, b.Area))

                                        Dim cc As CvContourChainCode = b.Contour
                                        cc.Render(imgContour)

                                        Dim polygon As CvContourPolygon = cc.ConvertToPolygon()
                                        For Each p As CvPoint In polygon
                                            imgPolygon.Circle(p, 1, CvColor.Red, -1)
                                        Next p
                                    Next item

                                    blobs.RenderBlobs(imgSrc, imgRender)

                                    Using TempCvWindow As CvWindow = New CvWindow("render", imgRender)
                                        Using TempCvWindowSrc As CvWindow = New CvWindow("contour", imgContour)
                                            Using TempCvWindowDest As CvWindow = New CvWindow("polygon vertices", imgPolygon)
                                                Cv.WaitKey(0)
                                            End Using
                                        End Using
                                    End Using
                                End Using
                            End Using
                        End Using
                    End Using
                End Using
            End Using
        End Sub
    End Module
' End Namespace
